pages/index.js

```diff
@@ -1,31 +1,43 @@
 import Head from 'next/head'
 import Image from 'next/image'
 import styles from '../styles/Home.module.css'
+import countries from '../lib/countries'
 
-export default function Home() {
+export const getServerSideProps = ({ query }) => ({
+  props: query,
+})
+
+export default function Home({ country }) {
   return (
     <div className={styles.container}>
       <Head>
-        <title>Next.js Edge Middleware on Netlify</title>
+        <title>Next.js Geolocation in Edge Functions</title>
         <meta
           name="description"
-          content="A simple starter template to demonstrates Edge Functions"
+          content="A simple starter template demonstrating Geolocation in Edge Functions"
         />
       </Head>
 
       <main className={styles.main}>
-        <h1 className={styles.title}>Next.js Edge Middleware on Netlify</h1>
-
-        <p className={styles.description}>
-          This is a simple starter template that demonstrates Edge Functions by
-          adding headers to the response.
-        </p>
-
+        <h1 className={styles.title}>Next.js Geolocation in Edge Functions</h1>
         <p className={styles.description}>
-          Get started by visiting{' '}
-          <a href="/shows/123">
-            <code className={styles.code}>shows/123</code>
-          </a>
+          {country && country !== 'undefined' ? (
+            <>
+              <span className={styles.flag}>
+                <Image
+                  alt={`${country} flag`}
+                  src={`https://flagcdn.com/96x72/${country.toLowerCase()}.png`}
+                  width={32}
+                  height={24}
+                />
+              </span>
+              <span className={styles.welcome}>
+                Welcome from {countries.find((c) => c.code === country).name}
```

Comment:
The silly thing is that the geo object that we pass an Edge Function already includes the country name, but Next.js doesn't expose it!

Reply:
Gah!